src/main/kotlin/com/exactpro/th2/check1/rule/RuleFactory.kt

```diff
@@ -272,10 +272,16 @@ class RuleFactory(
         }
     }
 
-    private fun checkCheckpoint(checkpoint: Checkpoint, sessionKey: SessionKey) {
-        check(checkpoint !== Checkpoint.getDefaultInstance()) { "Request doesn't contain a checkpoint" }
+    private fun checkCheckpoint(requestAdaptor: RequestAdaptor, sessionKey: SessionKey, isChainIdExist: Boolean) {
+        if (requestAdaptor.hasChainId) {
+            check(isChainIdExist) {
+                "The request has an invalid chain id or connectivity id"
+            }
+            return // We should validate checkpoint only if the request doesn't contain a chain id
+        }
+        check(requestAdaptor.hasCheckpoint) { "Request doesn't contain a checkpoint" }
```

Comment:
Please, add information about why the checkpoint is required (the messageTimeout is used and no chain ID was provided)